SimG4CMS/Calo/interface/HCalSD.h

```diff
@@ -48,6 +48,7 @@ class HCalSD : public CaloSD, public Observer<const BeginOfJob *> {
 
   double                getEnergyDeposit(const G4Step*) override;
   bool                  getFromLibrary(const G4Step*) override;
+  using CaloSD::update;
```

Comment:
@Dr15Jones , sorry, I may be miss the purpose of "using CaloSD::update", but would it be more simple (and correct) if instead of this construction we add into CaloSD.h 
void update(const BeginOfJob*) override; 
and empty implementation in CaloSD.cc?

Reply:
The `using` statement avoids hiding the `CaloSD::update` method from any code that is calling HCalSD. The clang compiler was complaining.

I disagree with your solution. Adding into CaloSD a method it doesn't want and one that we are adding just for an inheriting class is not a good maintenance idea. If, in the future, another class inherited from CaloSD you'd have to change CaloSD again.

What I did is the recommended modern way of doing this in C++.